src/coreclr/jit/morph.cpp

```diff
@@ -14534,6 +14534,38 @@ GenTree* Compiler::fgMorphMultiOp(GenTreeMultiOp* multiOp)
     if (opts.OptimizationEnabled() && multiOp->OperIs(GT_HWINTRINSIC))
     {
         GenTreeHWIntrinsic* hw = multiOp->AsHWIntrinsic();
+
+        // Move constant vectors from op1 to op2 for commutative and compare operations
+        // For now we only do it for zero vector
+        if (fgGlobalMorph && (HWIntrinsicInfo::lookupNumArgs(hw->GetHWIntrinsicId()) == 2) && hw->Op(1)->IsVectorZero())
```

Comment:
Why `fgGlobalMorph`? Commutative reordering is fine VN-wise.

Also, why `lookupNumArgs` instead of `GetOperandCount()`?